src/Server/StaticRequestHandler.cpp

```diff
@@ -175,7 +175,7 @@ HTTPRequestHandlerFactoryPtr createStaticHandlerFactory(IServer & server, const
     std::string response_content = server.config().getRawString(config_prefix + ".handler.response_content", "Ok.\n");
     std::string response_content_type = server.config().getString(config_prefix + ".handler.content_type", "text/plain; charset=UTF-8");
     auto factory = std::make_shared<HandlingRuleHTTPHandlerFactory<StaticRequestHandler>>(
-        server, std::move(response_content), std::move(status), std::move(response_content_type));
```

Comment:
Removing of `std::move` here leads that variadic args in [this](https://github.com/ClickHouse/ClickHouse/blob/91c8951a1a5441346c7b8d3a88e02ab8ac22d2fb/src/Server/HTTPHandlerFactory.h#L45-L55) quite complex code are deduced as:
`args: std::__1::tuple<DB::IServer&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, int&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >`, which leads to bug, because of reference to local int variable.

Reply:
Will take care of that, thanks.

How did you find the bug?